euphonic/qpoint_phonon_modes.py

```diff
@@ -169,19 +172,37 @@ def reorder_frequencies(self, reorder_gamma=True):
             evec_tmp = np.copy(self.eigenvectors[i, mode_map[i]])
             self.eigenvectors[i] = evec_tmp
 
-    def calculate_structure_factor(self, scattering_lengths, dw=None):
+    def calculate_structure_factor(self,
+        scattering_lengths: Union[str, Dict[str, Quantity]] = 'Sears1992',
+        dw: Optional[DebyeWaller] = None,
+        ) -> StructureFactor:
         """
         Calculate the one phonon inelastic scattering for neutrons at
         each q-point
 
         Parameters
         ----------
-        scattering_lengths : dictionary of float Quantity
-            Dictionary of spin and isotope averaged coherent scattering
-            length for each element in the structure in e.g.
-            {'O': 5.803*ureg('fm'), 'Zn': 5.680*ureg('fm')}
+        scattering_lengths : Dataset of coherent scattering length for each
+            element in the structure. This may be provided in 3 ways:
```

Comment:
The formatting of this line causes Sphinx to display it weirdly (see image). Since we've got type hints now maybe just put something like

```
scattering lengths
    This is a dataset of coherent scattering lengths for each element...
```

Then for consistency we'd also have to change the `dw` parameter to match. Perhaps I need to think about putting in type hints everywhere at some point...

![image](https://user-images.githubusercontent.com/18187336/89158953-b9defa80-d566-11ea-841a-7e157b2b104b.png)